cpp/arcticdb/version/test/test_version_map.cpp

```diff
@@ -547,56 +547,65 @@ TEST(VersionMap, StorageLogging) {
+
+    VersionId version_id { 0 };
+};
+
+std::shared_ptr<VersionMapEntry> write_versions(
+    const std::shared_ptr<InMemoryStore>& store,
+    const std::shared_ptr<VersionMap>& version_map,
+    const StreamId& id,
+    const std::vector<VersionChainOperation>& operations) {
     auto entry = version_map->check_reload(
             store,
             id,
             LoadStrategy{LoadType::NOT_LOADED, LoadObjective::INCLUDE_DELETED},
             __FUNCTION__);
 
-    auto key1 = atom_key_with_version(id, 0, 0);
-    version_map->do_write(store, key1, entry);
-    write_symbol_ref(store, key1, std::nullopt, entry->head_.value());
-    auto key2 = atom_key_with_version(id, 1, 1);
-    version_map->do_write(store, key2, entry);
-    // We override the symbol ref without a prev_key on purpose. This way we'll only load the version=1 from the ref key
-    write_symbol_ref(store, key2, std::nullopt, entry->head_.value());
+    for (const auto& [type, version_id]: operations) {
+        switch (type) {
+            case VersionChainOperation::Type::WRITE: {
+                auto key = atom_key_with_version(id, version_id, version_id);
+                version_map->do_write(store, key, entry);
+                write_symbol_ref(store, key, std::nullopt, entry->head_.value());
+                break;
+            }
+            case VersionChainOperation::Type::TOMBSTONE: {
+                version_map->write_tombstone(store, version_id, id, entry);
+                break;
+            }
+            case VersionChainOperation::Type::TOMBSTONE_ALL: {
+                version_map->delete_all_versions(store, id);
+                break;
+            }
+        }
+    }
 
     return entry;
 }
 
 // Produces the following version chain: v0 <- tombstone_all <- v1 <- v2 <- tombstone
 void write_alternating_deleted_undeleted(std::shared_ptr<InMemoryStore> store, std::shared_ptr<VersionMap> version_map, StreamId id) {
-    auto entry = version_map->check_reload(
-            store,
-            id,
-            LoadStrategy{LoadType::NOT_LOADED, LoadObjective::INCLUDE_DELETED},
-            __FUNCTION__);
-
-    auto key1 = atom_key_with_version(id, 0, 0);
-    auto key2 = atom_key_with_version(id, 1, 1);
-    auto key3 = atom_key_with_version(id, 2, 2);
-
-    // Write version 0
-    version_map->do_write(store, key1, entry);
-    write_symbol_ref(store, key1, std::nullopt, entry->head_.value());
-
-    // Tombstone_all on version 0
-    version_map->delete_all_versions(store, id);
-
-    // Write version 1
-    version_map->do_write(store, key2, entry);
-    write_symbol_ref(store, key2, std::nullopt, entry->head_.value());
-
-    // Write version 2
-    version_map->do_write(store, key3, entry);
-    write_symbol_ref(store, key3, std::nullopt, entry->head_.value());
+    using Type = VersionChainOperation::Type;
+    write_versions(store, version_map, id, {{Type::WRITE, 0},
+                                                         {Type::TOMBSTONE_ALL},
```

Comment:
Nit: These extra whitespaces seem off. Maybe align all the operations the same number of spaces to the right.

Hopefully @poodlewars's formatting will get merged and we can all use the same formatting throughout.